src/main/minio.js

```diff
@@ -2280,6 +2280,246 @@ export class Client {
+  setTagging(taggingParams){
+
+    const { bucketName, objectName, tags, putOpts={}, cb} = taggingParams
+    const method = 'PUT'
+    let query ="tagging"
+
+    if(putOpts && putOpts.versionId){
+      query =`${query}&versionId=${putOpts.versionId}`
+    }
+    const tagsList=[]
+    for (const [key, value] of Object.entries(tags)) {
+      tagsList.push( { Key: key, Value: value} )
+    }
+    const taggingConfig = {
+      Tagging: {
+        TagSet: {
+          Tag: tagsList
+        }
+      }
+    }
+    const encoder = new util.TextEncoder()
+    const headers ={}
+    const builder = new xml2js.Builder({ headless:true,renderOpts:{'pretty':false},})
+    let payload = builder.buildObject(taggingConfig)
+    payload = encoder.encode(payload)
+    const md5digest = Crypto.createHash('md5').update(payload).digest()
+
+    const requestOptions = { method, bucketName, query, headers }
+
+    if(objectName){
+      requestOptions['objectName']=objectName
+    }
+    headers['Content-MD5'] = md5digest.toString('base64')
+
+    this.makeRequest(requestOptions, payload, 200, '', false, cb)
+
+  }
+
+  /** Set Tags on a Bucket
+   * __Arguments__
+   * bucketName _string_
+   * tags _object_ of the form {'<tag-key-1>':'<tag-value-1>','<tag-key-2>':'<tag-value-2>'}
+   * `cb(error)` _function_ - callback function with `err` as the error argument. `err` is null if the operation is successful.
+   */
+  setBucketTagging(bucketName,tags,cb){
+    if (!isValidBucketName(bucketName)) {
+      throw new errors.InvalidBucketNameError('Invalid bucket name: ' + bucketName)
+    }
+    if(!isObject(tags)){
+      throw new errors.InvalidArgumentError('tags should be of type "object"')
+    }
+    if(Object.keys(tags).length > 50){
+      throw new errors.InvalidArgumentError('maximum tags allowed is 50"')
+    }
+    if (!isFunction(cb)) {
+      throw new errors.InvalidArgumentError('callback should be of type "function"')
+    }
+
+    return this.setTagging({bucketName, tags, cb})
+  }
+
+  /** Set Tags on an Object
+   * __Arguments__
+   * bucketName _string_
+   * objectName _string_
+   *  * tags _object_ of the form {'<tag-key-1>':'<tag-value-1>','<tag-key-2>':'<tag-value-2>'}
+   *  putOpts _object_ (Optional) e.g {versionId:"my-object-version-id"},
+   * `cb(error)` _function_ - callback function with `err` as the error argument. `err` is null if the operation is successful.
+   */
+  setObjectTagging(bucketName, objectName, tags, putOpts={}, cb){
+    if (!isValidBucketName(bucketName)) {
+      throw new errors.InvalidBucketNameError('Invalid bucket name: ' + bucketName)
+    }
+    if (!isValidObjectName(objectName)) {
+      throw new errors.InvalidBucketNameError('Invalid object name: ' + objectName)
+    }
+
+    if(isFunction(putOpts)){
+      cb=putOpts
+      putOpts={}
+    }
+
+    if(!isObject(tags)){
+      throw new errors.InvalidArgumentError('tags should be of type "object"')
+    }
+    if(Object.keys(tags).length > 50){
```

Comment:
https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObjectTagging.html

The limit is 10

Reply:
I have updated. Thank you.